### ISSUE TITLE
RuntimeError: Failed to import transformers.models.qwen2.modeling_qwen2 because of the following error (look up to see its traceback): Failed to import transformers.integrations.peft because of the following error (look up to see its traceback): No module named 'torch.distributed.checkpoint.format_utils'

### ISSUE THREAD

**Issue by imrankh46** [QwenLM/Qwen1.5#413]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

RuntimeError: Failed to import transformers.models.qwen2.modeling_qwen2 because of the following error (look up to see its traceback):
Failed to import transformers.integrations.peft because of the following error (look up to see its traceback):
No module named 'torch.distributed.checkpoint.format_utils'

### 期望行为 | Expected Behavior

installl transformers from the source and suggested version and also install attention but the error is the same.

### 运行环境 | Environment

```Markdown
- OS: colab
- Python:3.10
- Transformers: 4.40.2, 4.39.0,
- PyTorch: 2.2.1+cu121
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):
```

**Comment by jklj077** [comment]
Hi, 

`torch.distributed.checkpoint.format_utils` is not available until torch 2.3, but I cannot find it anywhere in transformers or peft code. It seems to be a colab thing. Is there a minimal colab notebook we can look into?

**Comment by jklj077** [comment]
I've discovered the source of the issue. The problem lies in the main branch of the `accelerate` library, specifically on this line: <https://github.com/huggingface/accelerate/blob/4ba436eccc1f6437503e66474d8ca86292f4acc1/src/accelerate/utils/fsdp_utils.py#L30>. However, it's worth noting that this particular change hasn't been included in any official release yet. Were you by any chance installing `accelerate` directly from the GitHub main branch instead of using an official release?
